src/shogun/multiclass/KNN.cpp

```diff
@@ -277,6 +288,25 @@ CMulticlassLabels* CKNN::apply_multiclass(CFeatures* data)
 #ifdef BENCHMARK_KNN
 		SG_PRINT(">>>> JL applied in %9.4f\n", tstart.cur_time_diff(false))
 #endif
+		break;
+	}
+	case KNN_KDTREE:
+	{
+		m_kd_tree->query_knn(dynamic_cast<CDenseFeatures<float64_t>*>(data), m_k);
+		SGMatrix<index_t> NN = m_kd_tree->get_knn_indices();
+		for (int32_t i=0; i<num_lab && (!CSignal::cancel_computations()); i++)
```

Comment:
This thing is actually quite good to have. We should put this into other algos.
Wanna send a snother patch for that?
I think it doesnt work for openmp though